template/index.js

```diff
@@ -0,0 +1,58 @@
+import { File } from "@asyncapi/generator-react-sdk";
+
+// Import custom components from file
+import { HTML, Head, Body } from "../components/common";
+
+export default function Index({ asyncapi, params }) {
+  const getCssLinks = () => {
+    const cssLinks = [];
+    if (params.singleFile) {
+      cssLinks.push("template/css/asyncapi-ui.css");
+      cssLinks.push("template/css/asyncapi-ui.css");
+    } else {
+      cssLinks.push("css/asyncapi-ui.css");
+      cssLinks.push("css/asyncapi-ui.css");
+    }
+    return cssLinks;
+  };
+  return (
+    <File name="index.html">
+      <HTML>
+        <Head
+          title={
+            asyncapi.info().title() +
+            asyncapi.info().version() +
+            "documentation"
+          }
+          cssLinks={getCssLinks()}
+          base={params.baseHref ? params.baseHref : ""}
+        />
+        <Body>
+          <div id="root">{asyncapi}</div>
```

Comment:
You should pass here rendered React component not AsyncAPI spec :) 

Original template -> https://github.com/asyncapi/html-template/blob/master/template/index.html#L22

You should use https://github.com/asyncapi/html-template/blob/master/filters/all.js#L81 function (pass params to it).